plugins/optimization-detective/optimization.php

```diff
@@ -253,6 +253,77 @@ function od_optimize_template_output_buffer( string $buffer ): string {
 	/**
 	 * Fires to register tag visitors before walking over the document to perform optimizations.
 	 *
+	 * Once a page has finished rendering and the output buffer is processed, the page contents are loaded into
```

Comment:
Not opposed to this block for now, but this is so detailed I think it should rather be part of some `docs/*.md` file. Maybe you can move (most of) it as part of #1857. This is probably relevant for one of the more detailed documentation articles to write, and we wouldn't want to have this much content duplicated between method docs and separate documentation.

Reply:
Yeah, as part of #1857 I'd love to have a process that would automatically extract the docs from phpdoc and dump it into the markdown files. That would ensure the documentation is as close to the code as possible (in PHP) while also being accessible for ease of reading (in Markdown).

Reply:
While a code reference like that would be helpful, I think large text chunks like this one here still fit better into a more detailed article, e.g. on how to write an Optimization Detective extension.

We'd probably want to avoid duplicating too much between those "manual" articles and automatically generated code reference as well.

Reply:
I've moved the inline docs out of phpdoc and into `hooks.md` in https://github.com/WordPress/performance/pull/1869

But as mentioned on that PR, I also have shared a drafted article for replacing what is now in `introduction.md`.